instrumentation/methods/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/methods/MethodInstrumentationModule.java

```diff
@@ -46,7 +57,9 @@ public List<String> getAdditionalHelperClassNames() {
     return typeInstrumentations.isEmpty()
         ? emptyList()
         : Arrays.asList(
+            "io.opentelemetry.javaagent.instrumentation.methods.MethodAndType",
             "io.opentelemetry.javaagent.instrumentation.methods.MethodSingletons",
+            "io.opentelemetry.javaagent.instrumentation.methods.MethodSingletons$1",
```

Comment:
Perhaps it is time to give up on this approach. You could solve the issue by ensuring there is always at least 1 element in `typeInstrumentations` list. If the list s empty you could add a `MethodInstrumentation` where class name is `null`. Then you can have to modify `MethodInstrumentation` to return `none()` in `typeMatcher` and `classLoaderOptimization` when the class name is `null`. This should let you delete this method.

Reply:
I think it is a small price for simplifying this. Since we need to return something only when running muzzle an alternative would be to somehow figure out when muzzle is checking the dependencies and only then make the list non-empty.